lib/coney/consumer_server.ex

```diff
@@ -33,8 +35,32 @@ defmodule Coney.ConsumerServer do
       ) do
     task = ExecutionTask.build(consumer, chan, payload, tag, meta)
 
-    spawn(ConsumerExecutor, :consume, [task])
+    {_pid, ref} = spawn_monitor(ConsumerExecutor, :consume, [task])
+
+    state = put_in(state.tasks[ref], %{chan: chan, tag: tag})
+
+    {:noreply, state}
+  end
+
+  # Received after the task completed successfully
+  def handle_info({:DOWN, ref, _, _, :normal = _reason}, state) do
+    {_task, state} = pop_in(state.tasks[ref])
+    Process.demonitor(ref, [:flush])
 
     {:noreply, state}
   end
+
+  # Received if the task terminate abnormally
+  def handle_info({:DOWN, ref, _, _, reason}, state) do
+    Logger.info("[#{__MODULE__}] Error processing message with reason: #{inspect(reason)}")
```

Comment:
does this have to be an info log, or can we change it to warning?

Reply:
If it terminates abnormally then it should be `error` no?

Reply:
We can change it to error, but for some projects, if there are frequent crashes during parsing, it will result in a high volume of error messages.